frontend/src/shape/molds/StatMold.ts

```diff
@@ -34,3 +38,18 @@ export class StatMold extends ShapeMold<StatShape> {
         this.statRowMold.assert(obj["rows"]);
     }
 }
+
+export function assertGenericStatShape(obj: any): StatShape {
+    assertKeyInShape(obj, "type");
+    assertElementOf(obj["type"], StatShapeTypes);
+
+    switch(obj["type"]) {
+        case "bar": return new BarStatMold().assert(obj);
+        case "line": return new LineStatMold().assert(obj);
+        case "bubble": return new BubbleStatMold().assert(obj);
+        case "histogram": return new HistogramStatMold().assert(obj);
+    }
+
+    // impossible error, but we need to include this line to make typescript happy
```

Comment:
i think youre the one who turned on the exhaustiveness check, no? or is it on by default?

Reply:
well, it's technically correct for it to be enabled, this is an edge case that could be fixed if i was a smarter programmer